File: Advanced/PageAndRDLX/CustomChart/VB.NET/TestViewer/ViewerForm.vb
Imports System.Text

Public Class ViewerForm

	Public Sub New()
		Encoding.RegisterProvider(CodePagesEncodingProvider.Instance)
		' This call is required by the designer.
		InitializeComponent()
	End Sub
	
	Protected Overrides Sub OnLoad(e As EventArgs)
		MyBase.OnLoad(e)
		viewer1.LoadDocument("..\..\..\..\..\..\Report\Radar.rdlx")
	End Sub
End Class

File: API/Section/Inheritance/C#/Inheritance/ViewerForm.cs
using System;
using System.Text;
using System.Windows.Forms;

namespace ActiveReports.Samples.Inheritance
{
	/// <summary>
	/// A description of the overview of the ViewerForm.
	/// </summary>
	public partial class ViewerForm : System.Windows.Forms.Form
	{
		public ViewerForm()
		{
			//
			//Required for Windows Form Designer support
			//
			InitializeComponent();
		 }

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main()
		{
			Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
			Application.EnableVisualStyles();
			Application.SetCompatibleTextRenderingDefault(false);
#if NET6_0_OR_GREATER
			Application.SetHighDpiMode(HighDpiMode.SystemAware);
#endif
			Application.Run(new ViewerForm());
		}

		private void Button1_Click(object sender, EventArgs e)
		{
			rptInheritChild rpt = new rptInheritChild();
			arvMain.LoadDocument(rpt);
		}

		private void Button2_Click(object sender, EventArgs e)
		{
			rptDesignChild rpt = new rptDesignChild();
			arvMain.LoadDocument(rpt);
		}
	}
}
